File: Manuel/개인 활동/Class 2/큐/queue.cpp
#include <iostream>
#include <string>
#include <queue>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    queue<int> q;
    
    for(int i = 0; i < N; i++)
    {
        string command;
        cin >> command;

        if(command == "push")
        {
            int X;
            cin >> X;
            q.push(X);
        }
        else if(command == "pop")
        {
            if(q.empty())
            {
                cout << -1 << "\n";
            }
            else
            {
                cout << q.front() << "\n";
                q.pop();
            }
        }
        else if(command == "size")
        {
            cout << q.size() << "\n";
        }
        else if(command == "empty")
        {
            if(q.empty())
            {
                cout << 1 << "\n";
            }
            else
            {
                cout << 0 << "\n";
            }
        }
        else if(command == "front")
        {
            if(q.empty())
            {
                cout << -1 << "\n";
            }
            else
            {
                cout << q.front() << "\n";
            }
        }
        else if(command == "back")
        {
            if(q.empty())
            {
                cout << -1 << "\n";
            }
            else
            {
                cout << q.back() << "\n";
            }
        }
    }
    
    return 0;
}

File: Manuel/개인 활동/Class 2/큐/queue.py
import sys
from collections import deque

def solve():
    N = int(sys.stdin.readline())
    queue = deque()

    for _ in range(N):
        command = sys.stdin.readline().split()

        if command[0] == "push":
            X = int(command[1])
            queue.append(X)
        elif command[0] == "pop":
            if queue:
                print(queue.popleft())
            else:
                print(-1)
        elif command[0] == "size":
            print(len(queue))
        elif command[0] == "empty":
            if not queue:
                print(1)
            else:
                print(0)
        elif command[0] == "front":
            if queue:
                print(queue[0])
            else:
                print(-1)
        elif command[0] == "back":
            if queue:
                print(queue[-1])
            else:
                print(-1)

solve()